packages/x-date-pickers/src/TimeClock/Clock.tsx

```diff
@@ -359,7 +360,9 @@ export function Clock<TDate>(inProps: ClockProps<TDate>) {
             ownerState={ownerState}
             className={classes.amButton}
           >
-            <Typography variant="caption">AM</Typography>
+            <Typography variant="caption" minWidth={19}>
```

Comment:
Without an explicit width, the icon button could become too narrow in certain locales (i.e. Arabic).
Do you have a better suggestion than forcing a `minWidth` to combat that? 🤔

Reply:
An overflow like we were proposing on the digital clock meridiem section?

Reply:
The relevant mentioned issue: https://github.com/mui/mui-x/issues/10031

Reply:
Adjusted to overflow text.
Before: https://codesandbox.io/s/zealous-estrela-chpfnl?file=/Demo.tsx

After: https://codesandbox.io/s/competent-leftpad-gvphzn?file=/Demo.tsx

Reply:
For me it make a lot of sense

Reply:
I like the idea of the ellipsis a lot! Makes sense and does not impact the design 🤌